tests/test_new_return_types.py

```diff
@@ -191,7 +191,45 @@ def circuit(x):
         assert res[0].shape == (2 ** len(wires),)
         assert isinstance(res[0], np.ndarray)
 
-    # Samples and counts
+    @pytest.mark.parametrize("measurement", [qml.sample(qml.PauliZ(0)), qml.sample(wires=[0])])
+    def test_sample(self, measurement):
+        """Test the sample measurement."""
+        shots = 1000
+        dev = qml.device("default.qubit", wires=2, shots=shots)
```

Comment:
General comment: we should add a fixture with device as suggested in PR1. I was abble to find a bug because of that.

Reply:
@rmoyard what made the difference in that case?

Reply:
The `execute_new` method is using `_asarray` which is related to the device and `default.mixed` function was not behaving correctly for floats.